hazelcast/src/main/java/com/hazelcast/jet/impl/execution/ConcurrentInboundEdgeStream.java

```diff
@@ -158,13 +161,23 @@ private static final class RoundRobinDrain extends InboundEdgeStreamBase {
             super(conveyor, ordinal, priority, debugName);
 
             this.waitForAllBarriers = waitForAllBarriers;
-            watermarkCoalescer = WatermarkCoalescer.create(conveyor.queueCount());
+            this.coalescers = new KeyedWatermarkCoalescer(conveyor.queueCount());
             receivedBarriers = new BitSet(conveyor.queueCount());
         }
 
         @Nonnull @Override
-        public ProgressState drainTo(@Nonnull Predicate<Object> dest) {
+        public ProgressState drainTo(@Nonnull Consumer<Object> dest) {
```

Comment:
It seems that we never return `DONE` state from this drainTo call() anymore. How do we handle the done state of CIES now?

Reply:
Yes, we don't return `DONE` explicitly. 
[Here](https://github.com/Fly-Style/hazelcast/blob/e25446dc262de3dbd1fcb0ce4964acb89047c5d0/hazelcast/src/main/java/com/hazelcast/jet/impl/execution/ConcurrentInboundEdgeStream.java#L249) we are returning a `ProgressState` assembled by a check if the last drained 'specical' item is `DONE_ITEM`. Upper in call stack `DONE_ITEM` is processed and the final `return` returns `DONE`.